### PR TITLE
Add M5Stack specific builds for esp32 light

### PR DESCRIPTION
**Pull request by andy31415**
This makes the M5Stack display work in case we are building for a m5stack.

A very large chung of this is actually taking existing UIs from all-clusters app and applying to m5stack.
Unfortunately I could see no modularization and created #25551 to fix that. I wanted to have a UI first and then modularize (mainly because I had to go through the entire process of figuring out Kconfig requirements to make all these work ... so now that UI compiles and works, may as well create a PR and not yak shave modularization).

Tested:
   - used `./scripts/build/build_examples.py --enable-flashbundle --target esp32-m5stack-light build` and verified display works
   - used `./scripts/build/build_examples.py --enable-flashbundle --target esp32-devkitc-light build` and flashed on a m5stack (assumed same mcu) and saw it working through logs (tested provision and toggle)

### CODE REVIEW

**Comment by andy31415** [comment]
Created #25551 to fix code duplication for M5Stack UIs ... we shold be able to more easily add UIs for examples than copy and paste (and I also see buttons and LEDs diverging quite a bit).

**Review by tcarmelveilleux — APPROVED** [review]

File: examples/lighting-app/esp32/main/Button.cpp
Comment (cliffamzn) [inline examples/lighting-app/esp32/main/Button.cpp:3]: if we're updating this, should we just make this 2023 too?
```suggestion
 *    Copyright (c) 2022-2023 Project CHIP Authors
```

File: examples/lighting-app/esp32/main/Button.cpp
Comment (cliffamzn) [inline examples/lighting-app/esp32/main/Button.cpp:105]:
```suggestion
                                (void *) (uint8_t) gpioNum, // init timer id = gpioNum index
```

File: examples/lighting-app/esp32/main/Button.cpp
Comment (cliffamzn) [inline examples/lighting-app/esp32/main/Button.cpp:115]: Do GPIOs of the M5 stack actually exceed 255? i don't think they do

```suggestion
    uint8_t gpio_num = (uint8_t) pvTimerGetTimerID(xTimer);
```

File: examples/lighting-app/esp32/main/DeviceCallbacks.cpp
Comment (cliffamzn) [inline examples/lighting-app/esp32/main/DeviceCallbacks.cpp:152]: does the M5 stack only support IPV4?

File: examples/lighting-app/esp32/main/DeviceWithDisplay.cpp
Comment (cliffamzn) [inline examples/lighting-app/esp32/main/DeviceWithDisplay.cpp:3]:
```suggestion
 *    Copyright (c) 2023 Project CHIP Authors
```

File: examples/lighting-app/esp32/main/DeviceWithDisplay.cpp
Comment (cliffamzn) [inline examples/lighting-app/esp32/main/DeviceWithDisplay.cpp:25]: should this go above line 19?

File: examples/lighting-app/esp32/main/Globals.cpp
Comment (cliffamzn) [inline examples/lighting-app/esp32/main/Globals.cpp:3]:
```suggestion
 *    Copyright (c) 2023 Project CHIP Authors
```

File: examples/lighting-app/esp32/main/StatusScreen.cpp
Comment (cliffamzn) [inline examples/lighting-app/esp32/main/StatusScreen.cpp:3]:
```suggestion
 *    Copyright (c) 2023 Project CHIP Authors
```

File: examples/lighting-app/esp32/main/include/Button.h
Comment (cliffamzn) [inline examples/lighting-app/esp32/main/include/Button.h:3]:
```suggestion
 *    Copyright (c) 2021-2023 Project CHIP Authors
```

File: examples/lighting-app/esp32/main/include/Button.h
Comment (cliffamzn) [inline examples/lighting-app/esp32/main/include/Button.h:45]: What's the point of providing the empty constructor? Won't it be useless if gpioNum isn't set?

File: examples/lighting-app/esp32/main/Button.cpp
Comment (andy31415) [reply examples/lighting-app/esp32/main/Button.cpp:115]: probably not ... however docs for pvTimerGetTimerId seem to return void* ... I actually only copied this code from existing places so did not review/write it ... it probably is awkward.

File: examples/lighting-app/esp32/main/DeviceCallbacks.cpp
Comment (andy31415) [reply examples/lighting-app/esp32/main/DeviceCallbacks.cpp:152]: When testing it seems it supports both. This was taken from what all-clusters-app does.

**Review by cliffamzn — APPROVED** [review]

File: examples/lighting-app/esp32/sdkconfig_m5stack.defaults
Comment (andy31415) [reply examples/lighting-app/esp32/sdkconfig_m5stack.defaults:39]: I would expect IPv4 and IPv6 configurations to be independent. At the same time, I grabbed this config from existing configurations assuming those were reasonable at some point.

File: examples/lighting-app/esp32/main/Button.cpp
Comment (andy31415) [reply examples/lighting-app/esp32/main/Button.cpp:97]: I believe you are correct, however I am afraid to miss something here - this was copied over from existing Button.cpp so I would prefer to keep it to have the files somewhat in sync for when we want to deduplicate them.

File: examples/lighting-app/esp32/main/Button.cpp
Comment (andy31415) [reply examples/lighting-app/esp32/main/Button.cpp:107]: I believe we should have a cleanup phase once we merge button.cpp (and other) implementations. Not diverging them will make things easier to merge around (even though some things already obviously diverged ... like different LED widget with different capabilities)

File: examples/lighting-app/esp32/main/include/Button.h
Comment (andy31415) [reply examples/lighting-app/esp32/main/include/Button.h:33]: From reading this, it seems there is an Init that can set a gpio number. My guess is empty constructors allows for array definitions.

The interface is awkward though - gpio number should probably be optional (or nullable in some way) and there seems to be a conflict of `button() + init()` or `button(gpio) + init(gpio)` combinations not making sense.

Keeping somewhat as is (but I removed the redundant comments) to have this consistent with other places. Would be nice to have a saner interface though.

**Pull request merged**
